docs/cellxgene_census_schema.md

```diff
@@ -872,8 +881,185 @@ Cell metadata MUST be encoded as a `SOMADataFrame` with the following columns:
+  </tr>
+  <tr>
+    <td>data</td>
+    <td>bool</td>
+    <td>It MUST be <code>True</code> if the scene contains spatial information about the oberservation, otherwise it MUST be <code>False</code>.</td>
+  </tr>
+  </tbody>
+</table>
+<!-- markdownlint-enable reference-links-images -->
+
+#### Positions array of a Scene – `census_obj["census_spatial_sequencing"][organism].spatial[scene_id].obsl["loc"]` – `PointCloudDataFrame`
+
+`scene_id` MUST correspond to the values `scene_id` in `census_obj["census_spatial_sequencing"][organism].obs_presence_matrix`.
+
+For each observation in each Scene, spatial array positions and additional positional metadata MUST be encoded as a `PointCloudDataFrame`.  Each row corresponds to an observation with the following columns:
+
+<!-- markdownlint-disable reference-links-images -->
+<table>
+<thead>
+  <tr>
+    <th>Column</th>
+    <th>Encoding</th>
+    <th>Description</th>
+  </tr>
+</thead>
+<tbody>
+  <tr>
+    <td>x</td>
+    <td>float</td>
+    <td>It MUST be the corresponding value in the <b>first</b> column of <code>obsm["spatial"]</code>. As defined in the CELLxGENE dataset schema.</td>
+  </tr>
+  </tr>
+    <td>y</td>
+    <td>float</td>
+    <td>It MUST be the corresponding value in the <b>second</b> column of <code>obsm["spatial"]</code>. As defined in the CELLxGENE dataset schema.</td>
+  </tr>
+  <tr>
+    <td>soma_joinid</td>
+    <td>integer</td>
+    <td>It MUST be valid <code>soma_joinid</code> from <code>census_obj["census_spatial_sequencing"][organism].obs</code>.</td>
+  </tr>
+  </tr>
+  </tr>
+ </tbody>
+</table>
+<!-- markdownlint-enable reference-links-images -->
+
+If Visium ("EFO:0010961") the units for the spatial array positions are pixels from the high-resolution image (`spatial[scene_soma_joinid].img[library_id]`). Otherwise TBD.
+
+The location dataframe MUST have the metadata fields:
+
+<!-- markdownlint-disable reference-links-images -->
+<table>
+<thead>
+  <tr>
+    <th>Key</th>
+    <th>Encoding</th>
+    <th>Description</th>
+  </tr>
+</thead>
+<tbody>
+  <tr>
+    <td>soma_geometry_type</td>
+    <td>str</td>
+    <td>MUST be "radius"</td>
+  </tr>
+  </tr>
+    <td>soma_geometry</td>
+    <td>float</td>
+    <td>Radius of points: <code>dimeter/2</code>. If Visium ("EFO:0010961") <code>diameter</code> MUST be <code>.uns["spatial"][library_id]['spot_diameter_fullres']</code>.  As defined in the CELLxGENE dataset schema. Otherwise TBD-TODO (else for Slide-seq it should be  0.003% of the radius occupied by the full cloud of points).</td>
+  </tr>
+ </tbody>
+</table>
+<!-- markdownlint-enable reference-links-images -->
+
+#### Images of a Scene - `census_obj["census_spatial_sequencing"][organism].spatial[scene_soma_joinid].img[library_id]` – `Collection` of `MultiscaleImage`
+
+Images of a Visium ("EFO:0010961") scene MUST adhere to the following specifications. Other assays MUST NOT have images, and MUST NOT include the `img` collection.
```

Comment:
You should be able to rely on / reuse the dataset schema requirement for uns.spatial[library_id] which guarantees that census builder receives validated data. For schema 5.2.0:

_Curator MUST annotate if assay_ontology_term_id is "EFO:0010961" for Visium Spatial Gene Expression and uns['spatial']['is_single'] is True; otherwise, this key MUST NOT be present._

So you could write:

This MUST only be present when there is a corresponding value in the source H5AD slot `.uns["spatial"][library_id]`, as defined in the CELLxGENE dataset schema.

Reply:
It's a little different here since we are not including the full resolution image that might be present in the original h5ad.